src/Package/Impl/DataInspect/Office/CsvAppFileIO.cs

```diff
@@ -46,36 +46,36 @@ internal static class CsvAppFileIO {
             var file = ProjectUtilities.GetUniqueFileName(folder, csvFileName, "csv", appendUnderscore: true);
             var rfile = file.Replace('\\', '/');
 
-            string currentStatusText = string.Empty;
-            await VsAppShell.Current.DispatchOnMainThreadAsync(() => {
-                var statusBar = VsAppShell.Current.GetGlobalService<IVsStatusbar>(typeof(SVsStatusbar));
-                statusBar.GetText(out currentStatusText);
-            });
+            string currentStatusText;
+            var statusBar = VsAppShell.Current.GetGlobalService<IVsStatusbar>(typeof(SVsStatusbar));
+            statusBar.GetText(out currentStatusText);
 
             try {
-                await SetStatusTextAsync(Resources.Status_WritingCSV);
-                using (var e = await session.BeginEvaluationAsync()) {
-                    await e.EvaluateAsync($"write.csv({result.Expression}, file='{rfile}')", REvaluationKind.Normal);
-                }
-
-                if (File.Exists(file)) {
-                    Task.Run(() => {
-                        try {
-                            Process.Start(file);
-                        } catch (Win32Exception ex) {
-                            ShowErrorMessage(ex.Message);
-                        } catch (FileNotFoundException ex) {
-                            ShowErrorMessage(ex.Message);
-                        }
-                    }).DoNotWait();
-                }
+                statusBar.SetText(Resources.Status_WritingCSV);
+                await CreateCsvAndStartProcess(result, session, rfile, file);
+            } catch (Win32Exception ex) {
+                VsAppShell.Current.ShowErrorMessage(string.Format(CultureInfo.InvariantCulture, Resources.Error_CannotOpenCsv, ex.Message));
+            } catch (FileNotFoundException ex) {
+                VsAppShell.Current.ShowErrorMessage(string.Format(CultureInfo.InvariantCulture, Resources.Error_CannotOpenCsv, ex.Message));
             } finally {
-                await SetStatusTextAsync(currentStatusText);
+                statusBar.SetText(currentStatusText);
             }
 
             Interlocked.Exchange(ref _busy, 0);
         }
 
+        private static async Task CreateCsvAndStartProcess(DebugEvaluationResult result, IRSession session, string rfile, string file) {
+            await TaskUtilities.SwitchToBackgroundThread();
+
+            using (var e = await session.BeginEvaluationAsync()) {
+                await e.EvaluateAsync($"write.csv({result.Expression}, file='{rfile}')", REvaluationKind.Normal);
```

Comment:
Side note: need to do a code review throughout the solution to find all places like this and replace with `ToRStringLiteral` everywhere.